server/src/db/schemas/compliance.ts

```diff
@@ -1,4 +1,9 @@
 import mongoose from 'mongoose';
+import { ComplianceReport } from 'myTypes';
+
+const isRequiredIfNotDraft = function(this: ComplianceReport, value: any) {
```

Comment:
change name of this function to suggest that it's form schema validator

Reply:
Okay, I will put there `validateRequiredIfNotDraftForForm` bit long but it gives a clearer understanding of the function's purpose and context.